gcore/gdal_typetraits.h

```diff
@@ -0,0 +1,510 @@
+    static constexpr OGRFieldType ogr_type = OFTReal;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_UInt64);
+    }
+};
+
+template <> struct CXXTypeTraits<float>
+{
+    static constexpr GDALDataType gdal_type = GDT_Float32;
+    static constexpr size_t size = sizeof(float);
+    static constexpr OGRFieldType ogr_type = OFTReal;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTFloat32;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_Float32);
+    }
+};
+
+template <> struct CXXTypeTraits<double>
+{
+    static constexpr GDALDataType gdal_type = GDT_Float64;
+    static constexpr size_t size = sizeof(double);
+    static constexpr OGRFieldType ogr_type = OFTReal;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_Float64);
+    }
+};
+
+template <> struct CXXTypeTraits<std::complex<float>>
+{
+    static constexpr GDALDataType gdal_type = GDT_CFloat32;
+    static constexpr size_t size = sizeof(float) * 2;
+    static constexpr OGRFieldType ogr_type = OFTMaxType;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_CFloat32);
+    }
+};
+
+template <> struct CXXTypeTraits<std::complex<double>>
+{
+    static constexpr GDALDataType gdal_type = GDT_CFloat64;
+    static constexpr size_t size = sizeof(double) * 2;
+    static constexpr OGRFieldType ogr_type = OFTMaxType;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_CFloat64);
+    }
+};
+
+#if defined(GDAL_ENABLE_FLOAT16) && defined(FLT16_MAX) && defined(FLT16_MIN)
+template <> struct CXXTypeTraits<_Float16>
+{
+    static constexpr GDALDataType gdal_type = GDT_Float16;
+    static constexpr size_t size = sizeof(_Float16);
+    static constexpr OGRFieldType ogr_type = OFTReal;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::Create(GDT_Float16);
+    }
+};
+#endif
+
+template <> struct CXXTypeTraits<std::string>
+{
+    static constexpr GDALDataType gdal_type = GDT_Unknown;
+    static constexpr size_t size = 0;
+    static constexpr OGRFieldType ogr_type = OFTString;
+    static constexpr OGRFieldSubType ogr_subtype = OFSTNone;
+
+    static inline GDALExtendedDataType GetExtendedDataType()
+    {
+        return GDALExtendedDataType::CreateString();
+    }
+};
+
+//! @endcond
+
+/** Trait accepting a GDALDaaType and mapping it to correspond C++ type and
```

Comment:
Missing a `t` in `GDALDataType`, and "correspond" should be "corresponding".

Reply:
fixed